packages/@lwc/ssr-compiler/src/compile-template/component.ts

```diff
@@ -52,7 +52,11 @@ function getChildAttrsOrProps(
                     : attr.value.value;
             return b.property('init', key, b.literal(value));
         } else if (attr.value.type === 'Literal' && typeof attr.value.value === 'boolean') {
-            return b.property('init', key, b.literal(attr.value.value));
+            return b.property(
+                'init',
+                key,
+                b.literal(attr.type === 'Attribute' ? '' : attr.value.value)
```

Comment:
Generates the following 
```js
    const childAttrs = {
      "data-attr": ""
    };
```
Instead of 
```js
    const childAttrs = {
      "data-attr": true
    };
```